test/functional/services/remote/remote.ts

```diff
@@ -50,6 +47,22 @@ export async function RemoteProvider({ getService }: FtrProviderContext) {
     }
   };
 
+  const writeCoverage = async (logEntry?: { message: string; level: string }) => {
+    const id = coverageCounter++;
+    const timestamp = Date.now();
+    const path = resolve(coverageDir, `${id}.${timestamp}.coverage.json`);
+    let coverageJson: string;
+    if (logEntry) {
+      const [, coverageJsonBase64] = logEntry.message.split(coveragePrefix);
+      coverageJson = Buffer.from(coverageJsonBase64, 'base64').toString('utf8');
+    } else {
+      const coverageObject = (await driver.executeScript('return window.__coverage__;')) as object;
```

Comment:
I think we should execute this logic in the cleanup hook rather than putting it here, and then pass the coverage to this function if the call to `driver.executeScript()` was successful and if the coverage is actually defined (we might be cleaning up before actually loading Kibana).

```ts
const coverageJson = await (
  driver.executeScript('return window.__coverage__')
    .catch(() => undefined)
    .then(coverage => coverage && JSON.stringify(coverage))
)

if (coverageJson) {
  await writeCoverage(coverageJson)
}
```